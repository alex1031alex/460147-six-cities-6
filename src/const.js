export const CardType = {
  CITY: `cities`,
  FAVORITE: `favorites`,
  NEARBY: `nearby`,
};

export const ImageSize = {
  STANDARD: {
    width: 260,
    height: 200
  },
  SMALL: {
    width: 150,
    height: 100
  }
};
