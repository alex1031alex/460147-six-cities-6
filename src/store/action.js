import {createAction} from '@reduxjs/toolkit';

export const ActionType = {
  CHANGE_CITY: `city/change`,
  REDIRECT_TO_ROUTE: `route/redirectToRoute`,
  REQUIRE_AUTHORIZATION: `user/requireAuthorization`,
  SET_AUTH_INFO: `user/setAuthInfo`,
  RESET_AUTH_INFO: `user/resetAuthInfo`,
  CHANGE_SORT_TYPE: `offers/changeSortType`,
  LOAD_OFFERS: `offers/loadOffers`,
  LOAD_REVIEWS: `offers/loadReviews`,
  LOAD_OFFER_BY_ID: `offers/loadOfferById`,
  LOAD_NEARBY_OFFERS: `offers/loadNearbyOffers`,
  UPDATE_OFFERS: `offers/updateOffers`,
  UPDATE_OFFER: `offers/updateOffer`,
  UPDATE_NEARBY_OFFERS: `offers/updateNearbyOffers`,
};

export const changeCity = createAction(ActionType.CHANGE_CITY);
export const changeSortType = createAction(ActionType.CHANGE_SORT_TYPE);
export const requireAuthorization = createAction(ActionType.REQUIRE_AUTHORIZATION);
export const loadOffers = createAction(ActionType.LOAD_OFFERS);
export const setAuthInfo = createAction(ActionType.SET_AUTH_INFO);
export const redirectToRoute = createAction(ActionType.REDIRECT_TO_ROUTE);
export const loadReviews = createAction(ActionType.LOAD_REVIEWS);
export const loadOfferById = createAction(ActionType.LOAD_OFFER_BY_ID);
export const loadNearbyOffers = createAction(ActionType.LOAD_NEARBY_OFFERS);
export const resetAuthInfo = createAction(ActionType.RESET_AUTH_INFO);
export const updateOffers = createAction(ActionType.UPDATE_OFFERS);
export const updateOffer = createAction(ActionType.UPDATE_OFFER);
export const updateNearbyOffers = createAction(ActionType.UPDATE_NEARBY_OFFERS);
