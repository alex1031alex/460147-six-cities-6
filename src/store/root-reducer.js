import {combineReducers} from 'redux';
import {city} from './city/city';
import {favorites} from './favorites/favorites';
import {offers} from './offers/offers';
import {user} from './user/user';

export const NameSpace = {
  CITY: `CITY`,
  OFFERS: `OFFERS`,
  USER: `USER`,
  FAVORITES: `FAVORITES`,
};

export default combineReducers({
  [NameSpace.CITY]: city,
  [NameSpace.OFFERS]: offers,
  [NameSpace.USER]: user,
  [NameSpace.FAVORITES]: favorites,
});
