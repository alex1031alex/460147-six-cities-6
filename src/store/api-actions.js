import {ActionCreator} from './action.js';
import {adaptOfferData, adaptOffersData, adaptAuthInfo, adaptReviews} from '../services/adapters.js';
import {AuthorizationStatus, ApiRoute, AppRoute} from '../const.js';

export const fetchOffers = () => (dispatch, _getState, api) => {
  api.get(ApiRoute.OFFERS)
    .then(({data}) => adaptOffersData(data))
    .then((data) => dispatch(ActionCreator.loadOffers(data)));
};

export const checkAuth = () => (dispath, _getState, api) => (
  api.get(ApiRoute.LOGIN)
    .then(() => dispath(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH)))
    .catch(() => {})
);

export const login = ({email, password}) => (dispatch, _getState, api) => (
  api.post(ApiRoute.LOGIN, {email, password})
    .then(({data}) => {
      dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      return data;
    })
    .then((data) => adaptAuthInfo(data))
    .then((data) => dispatch(ActionCreator.setAuthInfo(data)))
    .then(() => dispatch(ActionCreator.redirectToRoute(AppRoute.MAIN)))
);

export const fetchOfferById = (id) => (dispatch, _getState, api) => {
  return api.get(`/hotels/${id}`)
    .then(({data}) => adaptOfferData(data))
    .then((data) => dispatch(ActionCreator.loadOfferById(data)))
    .catch(() => dispatch(ActionCreator.redirectToRoute(ApiRoute.NOT_FOUND_PAGE)));
};

export const fetchReviews = (id) => (dispatch, _getState, api) => {
  return api.get(`/comments/${id}`)
    .then(({data}) => adaptReviews(data))
    .then((data) => dispatch(ActionCreator.loadReviews(data)))
    .catch(() => {});
};
