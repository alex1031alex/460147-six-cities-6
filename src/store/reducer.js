import {ActionType} from './action.js';
import {AuthorizationStatus, Cities, SortType} from '../const.js';

const initialState = {
  activeCity: Cities.PARIS,
  activeSortType: SortType.POPULAR,
  offers: [],
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  isOffersDataLoaded: false,
  authInfo: {},
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_CITY: {
      return {
        ...state,
        activeCity: action.payload
      };
    }

    case ActionType.CHANGE_SORT_TYPE: {
      return {
        ...state,
        activeSortType: action.payload
      };
    }

    case ActionType.REQUIRE_AUTHORIZATION: {
      return {
        ...state,
        authorizationStatus: action.payload
      };
    }

    case ActionType.LOAD_OFFERS: {
      return {
        ...state,
        offers: action.payload,
        isOffersDataLoaded: true,
      };
    }

    case ActionType.SET_AUTH_INFO: {
      return {
        ...state,
        authInfo: action.payload
      };
    }

    default: {
      return state;
    }
  }
};

export {reducer};
