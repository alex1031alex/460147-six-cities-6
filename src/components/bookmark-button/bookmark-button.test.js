import React from 'react';
import * as redux from 'react-redux';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';

import {NameSpace} from '../../store/root-reducer';
import {AuthStatus} from '../../const';

import BookmarkButton from './bookmark-button';

const mockStore = configureStore({});

it(`Bookmark button should render correctly`, () => {
  jest.spyOn(redux, `useSelector`);
  jest.spyOn(redux, `useDispatch`);

  const history = createMemoryHistory();
  const store = mockStore({
    [NameSpace.USER]: {
      authStatus: AuthStatus.AUTH,
    }
  });
  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <BookmarkButton
            isFavorite={true}
            parentClassName="place-card__bookmark-button"
            id={5}
          >
            <svg className="place-card__bookmark-icon" width={18} height={19}>
              <use xlinkHref="#icon-bookmark" />
            </svg>
          </BookmarkButton>
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
