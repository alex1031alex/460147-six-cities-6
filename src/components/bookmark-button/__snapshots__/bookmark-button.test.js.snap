// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmark button should render correctly 1`] = `
<div>
  <button
    class="button place-card__bookmark-button place-card__bookmark-button--active"
    type="button"
  >
    <svg
      class="place-card__bookmark-icon"
      height="19"
      width="18"
    >
      <use
        xlink:href="#icon-bookmark"
      />
    </svg>
    <span
      class="visually-hidden"
    >
      $
      In
       bookmarks
    </span>
  </button>
</div>
`;
